rust/src/delta.rs

```diff
@@ -1355,6 +1455,223 @@ fn log_entry_from_actions(actions: &[Action]) -> Result<String, serde_json::Erro
+        let last_checkpoint_content = serde_json::to_string(&last_checkpoint_content)?;
+
+        info!(
+            "Writing _last_checkpoint to {:?}.",
+            self.last_checkpoint_path
+        );
+        self.storage
+            .put_obj(
+                self.last_checkpoint_path.as_str(),
+                last_checkpoint_content.as_bytes(),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    fn parquet_bytes_from_state(
+        &self,
+        state: &DeltaTableState,
+    ) -> Result<Vec<u8>, CheckPointWriterError> {
+        let current_metadata = state
+            .current_metadata()
+            .ok_or(CheckPointWriterError::MissingMetaData)?;
+
+        let mut json_buffer: Vec<u8> = Vec::new();
+
+        let protocol = action::Action::protocol(action::Protocol {
+            min_reader_version: state.min_reader_version(),
+            min_writer_version: state.min_writer_version(),
+        });
+        extend_json_byte_buffer(&mut json_buffer, &protocol)?;
+
+        let metadata = action::Action::metaData(action::MetaData::try_from(current_metadata)?);
+        extend_json_byte_buffer(&mut json_buffer, &metadata)?;
+
+        for add in state.files() {
+            let add = action::Action::add(add.clone());
+            extend_json_byte_buffer(&mut json_buffer, &add)?;
+        }
+
+        for remove in state.tombstones() {
+            let remove = action::Action::remove(remove.clone());
+            extend_json_byte_buffer(&mut json_buffer, &remove)?;
+        }
+
+        for (app_id, version) in state.app_transaction_version().iter() {
+            let txn = action::Action::txn(action::Txn {
+                app_id: app_id.clone(),
+                version: *version,
+                last_updated: None,
+            });
+            extend_json_byte_buffer(&mut json_buffer, &txn)?;
+        }
+
+        let checkpoint_schema = self.schema_factory.delta_log_schema_for_table(
+            &current_metadata.schema,
+            current_metadata.partition_columns.as_slice(),
+        )?;
+        let arrow_checkpoint_schema: ArrowSchema =
+            <ArrowSchema as TryFrom<&Schema>>::try_from(&checkpoint_schema)?;
+        let arrow_schema = Arc::new(arrow_checkpoint_schema);
+
+        let cursor = Cursor::new(json_buffer);
+
+        let mut json_reader = ReaderBuilder::new()
+            .with_schema(arrow_schema.clone())
+            .build(cursor)?;
+
+        debug!("Preparing checkpoint parquet buffer.");
+
+        let writeable_cursor = InMemoryWriteableCursor::default();
+        let mut writer = ArrowWriter::try_new(writeable_cursor.clone(), arrow_schema, None)?;
+
+        debug!("Writing to checkpoint parquet buffer...");
+
+        while let Some(batch) = json_reader.next()? {
+            writer.write(&batch)?;
+        }
+
+        let _ = writer.close()?;
+
+        info!("Finished writing checkpoint file.");
```

Comment:
Since it's info, do you mind adding the file name/path to this log

Reply:
Whoops, that log statement is incorrect actually. The file isn't in scope here. This is just completion of the buffer. Fixed that and changed it to debug. The file info statement is [here](https://github.com/delta-io/delta-rs/pull/280/files#diff-3b5c80af9e5527fbeaa6dad151a835f0733a26e61d0ba0471c58f243a29dfa99R1569).

Reply:
<grumble />. apparently that link doesn't work. currently line 1569 of `delta.rs`.